Ast/src/Lexer.cpp

```diff
@@ -535,37 +587,67 @@ Lexeme Lexer::readQuotedString()
             return Lexeme(Location(start, position()), Lexeme::BrokenString);
 
         case '\\':
+            readBackslashInString();
+            break;
+
+        default:
             consume();
-            switch (peekch())
-            {
-            case '\r':
-                consume();
-                if (peekch() == '\n')
-                    consume();
-                break;
+        }
+    }
 
-            case 0:
-                break;
+    consume();
 
-            case 'z':
-                consume();
-                while (isSpace(peekch()))
-                    consume();
-                break;
+    return Lexeme(Location(start, position()), Lexeme::QuotedString, &buffer[startOffset], offset - startOffset - 1);
+}
 
-            default:
-                consume();
-            }
+Lexeme Lexer::readInterpolatedStringBegin()
+{
+    LUAU_ASSERT(peekch() == '`');
+
+    Position start = position();
+    consume();
+
+    return readInterpolatedStringSection(start, Lexeme::InterpStringBegin, Lexeme::InterpStringSimple);
+}
+
+Lexeme Lexer::readInterpolatedStringSection(Position start, Lexeme::Type formatType, Lexeme::Type endType)
+{
+    unsigned int startOffset = offset;
+
+    while (peekch() != '`')
+    {
+        switch (peekch())
+        {
+        case 0:
+        case '\r':
+        case '\n':
+            // INTERP TODO: Clear anything we've added to the brace stack, and write a test to see what happens if we don't
```

Comment:
This would be a good idea to actually clear, because you might have gotten yourself in a situation where you don't have a paired `` ` ``/`` }...` ``. Not doing this I think will pretty much turn every lexemes orange.

Indeed, a test to prove the above theory would be great. It even would serve as a reminder to always clear the stack.

Reply:
I think I'm going to add some state for number of interpolated strings we're in, and clear the stack when it reaches 0 (and don't add onto it if it is).

Reply:
Oh, will need to solve the issue where we don't have a matching \` either. I'll figure it out.

Reply:
Currently trying to get a test that fails, this works completely as expected (one error for the malformed string):

```lua
_ = `{a`
            _ = { 1, 2, 3 }
            _ = `some {"more"}`

Reply:
Got it, ``_ = { `a{` } `` causes problems.

Reply:
Actually, issue with that is because it's being interpreted as:


{
InterpStringBegin("a")
BrokenString(" }")

...which I don't see a straight forward solution for. Either way, I can't figure out a way to break the brace stack right now.